src/controllers/auth.js

```diff
@@ -0,0 +1,162 @@
+const bcrypt = require('bcrypt');
+const { Volunteer, Organization } = require('../models/user');
+
+exports.generalsignin = async (req, res) => {
+  const { email, password } = req.body;
+
+  const volunteer = await Volunteer.findOne({ email });
+  const organization = await Organization.findOne({ email });
+
+  const volunteerpassword = await Volunteer.findOne({ password });
+  const organizationpassword = await Organization.findOne({ password });
```

Comment:
- I would recommend adding User to model exports and using it to find all the users. 
- Since we have username inside the models I think it's more suitable to use username instead of email.
- You should also not query the database with the given password.
```suggestion
const user = await User.findOne({ username});
const passwordHash = user.password;
```